Metaschema XSLT Inspector (#73)
usnistgov/metaschema-xslt: src/schema-gen/InspectorXSLT/TESTING.md

- Functionality of the Inspector XSLT
- Interfaces and outputs / runtime options for Inspector XSLT

Currently we focus on the *first two* of these, producing functional results in a simplified format that we can build to later - testing interface targets at that time as appropriate. An example would be producing HTML or Markdown reports: for now we test only generic reports in a format we control.

Find resources for testing the XSLT Inspector and its production in the [testing](testing) subdirectory.

## Model metaschemas for testing

### `current` resource cache

The testing/current directory contains copies of resources produced by the Inspector generator and related tools for testing, including:

- Current-best Inspector implementation for any metaschemas, as generated
- Current-best XSD expressing (a subset of) the same rules as the corresponding Inspector, for a metaschema module, for testing
- Current-best 'composed' metaschema instance for each testing metaschema. i.e. a fully assembled and linked metaschema module, useful for debugging

Within these artifacts, initial comments should give information regarding date of creation.

### "Computer Model" metaschema

> This suite uses an extension of AJ Stein's 'computer metaschema' model for testing, maintained here as [testing/computer_metaschema.xml](testing/computer_metaschema.xml).

Reviewer: Minor, non-blocking: technically it's the Metaschema project's model. Dave and others have made significant improvements there.

```suggestion
This suite uses an extension of 'computer metaschema' model from [examples included in the specification's repository](https://github.com/usnistgov/metaschema/blob/develop/examples/) for testing, maintained here as [testing/computer_metaschema.xml](testing/computer_metaschema.xml).
```
